op-node/p2p/sync.go

```diff
@@ -528,34 +537,45 @@ func (s *SyncClient) peerLoop(ctx context.Context, id peer.ID) {
 		// once the peer is available, wait for a sync request.
 		select {
 		case pr := <-s.peerRequests:
+			if !s.activeRangeRequests[pr.rangeReqId] {
+				log.Debug("dropping cancelled p2p sync request", "num", pr.num)
+				delete(s.inFlight, pr.num)
+				continue
+			}
+
 			// We already established the peer is available w.r.t. rate-limiting,
 			// and this is the only loop over this peer, so we can request now.
 			start := time.Now()
+
+			resultCode := byte(0)
 			err := s.doRequest(ctx, id, pr.num)
 			if err != nil {
 				// mark as complete if there's an error: we are not sending any result and can complete immediately.
 				pr.complete.Store(true)
 				log.Warn("failed p2p sync request", "num", pr.num, "err", err)
+
+				if re, ok := err.(requestResultErr); ok {
+					resultCode = re.ResultCode()
+					if re.ResultCode() == 1 { // indicates block not found error
+						log.Warn("cancelling p2p sync range request", "reqId", pr.rangeReqId)
+						delete(s.activeRangeRequests, pr.rangeReqId)
+					}
+				} else {
+					resultCode = 1
+				}
+
 				s.appScorer.onResponseError(id)
```

Comment:
Changed code in https://github.com/ethereum-optimism/optimism/pull/10063/commits/1ee4c26fcc8fd0032d9aa2a83fc41b3868eb9a2a to prevent peer from being penalized for this specific error